src/index.js

```diff
@@ -1,32 +1,53 @@
-var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
+//var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
 
-var API = 'https://rickandmortyapi.com/api/character/';
+var API = "https://rickandmortyapi.com/api/character/";
 var xhttp = new XMLHttpRequest();
+//Resolución d elso primeros dos problemas
 
 function fetchData(url_api, callback) {
-  xhttp.onreadystatechange = function (event) {
-    if (xhttp.readyState === '4') {
-      if (xhttp.status == 200)
-        callback(null, xhttp.responseText);
-      else return callback(url_api);
+
+/*   function status(readyState) {
+    switch (readyState) {
+      case 0: return 'no inicializado'
+      case 1: return 'cargando...'
+      case 2: return 'cargado'
+      case 3: return 'interactivo'
+      case 4: return 'completado'        
     }
-  };
-  xhttp.open('GET', url_api, false);
+    
+  }
+ */
+
+  xhttp.open("GET", url_api, true);  
   xhttp.send();
-};
+  xhttp.onreadystatechange = even=> {
+    if (xhttp.readyState === 4) {
+      if (xhttp.status == 200) callback(null, JSON.parse(xhttp.responseText));
+      else return callback(url_api);
```

Comment:
Te recomiendo ser mas claros en el reject, esto te permite identificar el error mas fácil. 

```suggestion
          reject(new Error("Algo salió mal."));
```